packages/flutter/lib/src/material/about.dart

```diff
@@ -558,49 +891,100 @@ class _LicensePageState extends State<LicensePage> {
   @override
   Widget build(BuildContext context) {
     assert(debugCheckHasMaterialLocalizations(context));
-    final String name = widget.applicationName ?? _defaultApplicationName(context);
-    final String version = widget.applicationVersion ?? _defaultApplicationVersion(context);
-    final Widget icon = widget.applicationIcon ?? _defaultApplicationIcon(context);
     final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-    return Scaffold(
-      appBar: AppBar(
-        title: Text(localizations.licensesPageTitle),
-      ),
-      // All of the licenses page text is English. We don't want localized text
-      // or text direction.
-      body: Localizations.override(
-        locale: const Locale('en', 'US'),
-        context: context,
-        child: DefaultTextStyle(
-          style: Theme.of(context).textTheme.caption,
-          child: SafeArea(
-            bottom: false,
-            child: Scrollbar(
-              child: ListView(
-                padding: const EdgeInsets.symmetric(horizontal: 8.0, vertical: 12.0),
-                children: <Widget>[
-                  Text(name, style: Theme.of(context).textTheme.headline5, textAlign: TextAlign.center),
-                  if (icon != null) IconTheme(data: Theme.of(context).iconTheme, child: icon),
-                  Text(version, style: Theme.of(context).textTheme.bodyText2, textAlign: TextAlign.center),
-                  Container(height: 18.0),
-                  Text(widget.applicationLegalese ?? '', style: Theme.of(context).textTheme.caption, textAlign: TextAlign.center),
-                  Container(height: 18.0),
-                  Text('Powered by Flutter', style: Theme.of(context).textTheme.bodyText2, textAlign: TextAlign.center),
-                  Container(height: 24.0),
-                  ..._licenses,
-                  if (!_loaded)
-                    const Padding(
-                      padding: EdgeInsets.symmetric(vertical: 24.0),
-                      child: Center(
-                        child: CircularProgressIndicator(),
-                      ),
-                    ),
-                ],
+    final ThemeData theme = Theme.of(context);
+    final String title = widget.packageName;
+    final String subtitle = localizations.licensesPackageDetailText(widget.licenseEntries.length);
+    final double pad = _getGutterSize(context);
+    final EdgeInsets padding = EdgeInsets.only(left: pad, right: pad, bottom: pad);
+    final List<Widget> listWidgets = <Widget>[
+      ..._licenses,
+      if (!_loaded)
+        const Padding(
+          padding: EdgeInsets.symmetric(vertical: 24.0),
+          child: Center(
+            child: CircularProgressIndicator(),
+          ),
+        ),
+    ];
+
+    Widget page;
+    if (widget.scrollController == null) {
+      page = Scaffold(
+        appBar: AppBar(
+          title: _PackageLicensePageTitle(title, subtitle, theme.primaryTextTheme),
```

Comment:
@TonicArtos 

Why did you specify `theme.primaryTextTheme`?

It may break color system when `AppBarTheme` is used for example.

Reply:
Indeed, it broke my background color.

Reply:
I would have preferred for this to (private) widget to have been factored a little differently - to not have used primaryTextTheme.

If you can file an issue with a simple test case that demonstrates a regression, we'll try and fix it.

Reply:
@HansMuller 

> If you can file an issue with a simple test case that demonstrates a regression, we'll try and fix it.

Okay, I've created an issue: #63239.

Reply:
@HansMuller 

I tried to fix at #63249, but I’m welcome to be closed and you enhance the implementation.

Reply:
> I would have preferred for this to (private) widget to have been factored a little differently - to not have used primaryTextTheme.
> 
> If you can file an issue with a simple test case that demonstrates a regression, we'll try and fix it.
> 

Sorry about the issue. What do you think the original solution should have been?

Reply:
@TonicArtos 

> What do you think the original solution should have been?

I think #63239 is the simplest fix for exiting implementation.